### ISSUE TITLE
How do i get notified about FNC1 (232) symbols?

### ISSUE THREAD

**Issue by 5263** [dmtx/libdmtx#1]
How do i get notified about FNC1 (232) symbols?

**Comment by preexo** [comment]
+1 for this, is this possible? FNC1 character are quite important.

**Comment by msva** [comment]
Sorry for delay again :(
Unfortunatelly, I've deadlines in my job ATM, so I can't properly implement that. Although, yes, I think, it is quite important (although, not so many readers support that, anyway).

So, if anyone want to make it and send PR — I'll happily merge it.

**Comment by mungewell** [comment]
Can someone with GS1 knowledge explain how the reader knows the length of AI. If the AI is more than 2 digits how is this encoded into the codewords? Does reader 'just' need prior knowledge of all relevant AI and their sizes/format?

**Comment by mungewell** [comment]
Could someone with experience in GS-1 codes have a look at my fork:
https://github.com/mungewell/libdmtx/tree/gs1-attempt1

I believe that it is now encoding/decoding properly. The FNC1 character is output as the character defined by '-G xx'. For example:
![test5](https://user-images.githubusercontent.com/236907/37746923-274ac600-2d42-11e8-8d0a-931695f822cd.jpg)
```
$ find . -name '*.jpg' -exec bash -c "echo {};dmtxread -G 43 {} ; echo; dmtxread -G 43 {} -c | grep d ; echo" \;

./test5.jpg
+0010030378123456789017130700103033262+3000017
d:232
d:130
d:140
d:133
d:133
d:208
d:142
d:164
d:186
d:208
d:220
d:147
d:143
d:137
d:130
d:140
d:160
d:163
d:156
d:051
d:232
d:160
d:130
d:131
d:056
d:129
```
To be totally correct it should add brackets to output, but that would require knowing all AI codes and maybe allow specifying additional ones

Comments?

**Comment by msva** [comment]
> To be totally correct it should add brackets to output, but that would require knowing all AI codes and maybe allow specifying additional ones

Maybe, that should be done on the client application side, but not a library?

Anyway, thanks a lot (I'd say "A LOT") of your work.

P.S. @5263 I guess, commit comment (as well as previous comment) done by @mungewell do reply on your question, and issue is resolved now?

P.P.S. Semi-offtop: @mungewell what do you think about implementation of ECI in libdmtx/dmtxutils as well? :grinning: 

ref: https://en.wikipedia.org/wiki/Extended_Channel_Interpretation
ref: https://en.wikipedia.org/wiki/Data_Matrix#Encoding
ref: https://www.morovia.com/manuals/datamatrix-font-encoder5/Input-Format.php#mdf-input-format-sect-3

**Comment by minus7** [comment]
If anyone else is looking for the latest status on FNC1: it was merged in #10

**Comment by xjpf** [comment]
still we hope for fnc1 support :)

**Comment by msva** [comment]
> still we hope for fnc1 support for gs1 datamatrix generation :)

wasn't it implemented in #10 and merged long time ago?
